pkg/registry/apis/secret/decrypt/client.go

```diff
@@ -0,0 +1,52 @@
+package decrypt
+
+import (
+	"context"
+
+	"github.com/grafana/grafana/pkg/registry/apis/secret/contracts"
+	"github.com/grafana/grafana/pkg/registry/apis/secret/xkube"
+	"github.com/grafana/grafana/pkg/services/featuremgmt"
+)
+
+type DecryptClientFunc struct {
+	decryptStorage contracts.DecryptStorage
+}
+
+var _ contracts.DecryptClient = (*DecryptClientFunc)(nil)
+
+// ProvideDecryptClientFunc for services that want to decrypt secure values in single-tenant mode.
+func ProvideDecryptClientFunc(features featuremgmt.FeatureToggles, decryptStorage contracts.DecryptStorage) (contracts.DecryptClient, error) {
+	if !features.IsEnabledGlobally(featuremgmt.FlagGrafanaAPIServerWithExperimentalAPIs) ||
+		!features.IsEnabledGlobally(featuremgmt.FlagSecretsManagementAppPlatform) {
+		return &DecryptClientFunc{}, nil
+	}
+
+	return &DecryptClientFunc{
+		decryptStorage: decryptStorage,
+	}, nil
+}
+
+// Decrypt a series of `names` from a `namespace`.
+func (d *DecryptClientFunc) Decrypt(ctx context.Context, namespace xkube.Namespace, names ...string) (map[string]string, error) {
```

Comment:
what is `xkube.Namespace` -- is that just namespace?

Reply:
Yes, it's just a newtype over a string, to get stronger safety, in the past we passed the wrong thing because some methods accept name and namespace and they are both plain strings.